niftilib/nifti1_io.c

```diff
@@ -2441,7 +2441,7 @@ int nifti_get_filesize( const char *pathname )
 
    if( pathname == NULL || *pathname == '\0' ) return -1 ;
    ii = stat( pathname , &buf ); if( ii != 0 ) return -1 ;
-   return (unsigned int)buf.st_size ;
+   return (int)buf.st_size ;
```

Comment:
Nice catch!

Reply:
I think this `unsigned` was intentional.  From what I recall, some compilers get mad about changing both the sign and the size of an integer, and this may have been done as a 2-step process.
Maybe a better cast would be: (int)(unsigned int)

Reply:
I've dropped this change.